implementations/rust/ockam/ockam_api/src/cli_state/enrollments.rs

```diff
@@ -276,3 +277,466 @@ impl EnrollmentTicket {
+    pub one_time_code: OneTimeCode,
+    project_route: ProjectRoute,
+    project_name: String,
+    project_change_history: String,
+    authority_change_history: String,
+    authority_route: Option<String>,
+}
+
+impl ExportedEnrollmentTicket {
+    const MANDATORY_FIELDS_NUM: usize = 5;
+
+    pub fn new(
+        one_time_code: OneTimeCode,
+        project_route: ProjectRoute,
+        project_name: impl Into<String>,
+        project_change_history: impl Into<String>,
+        authority_change_history: impl Into<String>,
+        authority_route: Option<impl Into<String>>,
+    ) -> Self {
+        Self {
+            one_time_code,
+            project_route,
+            project_name: project_name.into(),
+            project_change_history: project_change_history.into(),
+            authority_change_history: authority_change_history.into(),
+            authority_route: authority_route.map(Into::into),
+        }
+    }
+
+    #[cfg(any(test, feature = "test-utils"))]
+    pub fn new_test() -> Self {
+        Self::new(
+            OneTimeCode::new(),
+            ProjectRoute::new_with_id("project_id").unwrap(),
+            crate::cli_state::random_name(),
+            "81825837830101583285f68200815820245ba33c7729dce1c94d8c1a00fcf89a7af33689d4563176f9dffbdd147d4488f41a66e2ee7b1a79aef17b820081584070856bb8da621154a39c894a2fedded55257715b00940b9cffe54b51d87889aff2c077124ee6e0e1c2e711688470affbc65d909c87acf4e41d38bdfb03e2000d",
+            "81825837830101583285f6820081582045d9dac79f226762025fc82e7407aee4a4c8e7068dc04edd44f1c777b8f0cf6bf41a66e2ee7b1a79aef17b8200815840c65ce655fd57cf2ea0b0679066a24bc99e2b223341186b5eaec951101f291e96c5fc8343291a23cbd8dc063ad1f9a9554f036e8f34ab5388e444977e7e29ab0b",
+            None::<String>,
+        )
+    }
+
+    pub async fn import(self) -> Result<EnrollmentTicket> {
+        EnrollmentTicket::new(
+            self.one_time_code,
+            self.project_route.id,
+            self.project_name,
+            self.project_route.route.to_string(),
+            self.project_change_history,
+            self.authority_change_history,
+            self.authority_route,
+        )
+        .await
+    }
+}
+
+impl FromStr for ExportedEnrollmentTicket {
+    type Err = ApiError;
+
+    fn from_str(contents: &str) -> std::result::Result<Self, Self::Err> {
+        // Decode as comma-separated text
+        let values: Vec<&str> = contents.split(',').collect();
+        if values.len() < Self::MANDATORY_FIELDS_NUM {
+            return Err(ApiError::core("Missing fields in enrollment ticket").into());
+        }
+        let (
+            project_id_or_route,
+            project_name,
+            one_time_code,
+            project_change_history,
+            authority_change_history,
+        ) = (values[0], values[1], values[2], values[3], values[4]);
+        let authority_route = values.get(5).map(|r| r.to_string());
+
+        let project_id_or_route = if project_id_or_route.starts_with('/') {
```

Comment:
I think we can call the variable in `let project_id_or_route`, simply `project_route` after we checked for `starts_with('/')`.